cpukit/score/src/schedulerstrongapa.c

```diff
@@ -3,339 +3,820 @@
-    &self->Bit_map,
-    &self->Ready[ 0 ]
-  );
+  Scheduler_strong_APA_Context *self;
+  Scheduler_strong_APA_Node    *node;
+
+  self = _Scheduler_strong_APA_Get_self( context );
+  node = _Scheduler_strong_APA_Node_downcast( node_to_extract );
+
+  _Scheduler_SMP_Extract_from_scheduled( &self->Base.Base, &node->Base.Base );
+  /* Not removing it from Ready since the node could go in the READY state */
 }
 
-static Scheduler_strong_APA_Context *
-_Scheduler_strong_APA_Get_context( const Scheduler_Control *scheduler )
+static inline void _Scheduler_strong_APA_Extract_from_ready(
+  Scheduler_Context *context,
+  Scheduler_Node    *node_to_extract
+)
 {
-  return (Scheduler_strong_APA_Context *) _Scheduler_Get_context( scheduler );
+  Scheduler_strong_APA_Context *self;
+  Scheduler_strong_APA_Node    *node;
+
+  self = _Scheduler_strong_APA_Get_self( context );
+  node = _Scheduler_strong_APA_Node_downcast( node_to_extract );
+
+  if( !_Chain_Is_empty( &self->Ready ) ) {
+    if( !_Chain_Is_node_off_chain( &node->Ready_node ) ) {
+      _Chain_Extract_unprotected( &node->Ready_node );
+      _Chain_Set_off_chain( &node->Ready_node );
+    }
+  }
 }
 
-void _Scheduler_strong_APA_Initialize( const Scheduler_Control *scheduler )
+static inline void _Scheduler_strong_APA_Insert_ready(
+  Scheduler_Context *context,
+  Scheduler_Node    *node_base,
+  Priority_Control   insert_priority
+)
 {
-  Scheduler_strong_APA_Context *self =
-    _Scheduler_strong_APA_Get_context( scheduler );
+  Scheduler_strong_APA_Context *self;
+  Scheduler_strong_APA_Node    *node;
 
-  _Scheduler_SMP_Initialize( &self->Base );
-  _Priority_bit_map_Initialize( &self->Bit_map );
-  _Scheduler_priority_Ready_queue_initialize(
-    &self->Ready[ 0 ],
-    scheduler->maximum_priority
-  );
+  self = _Scheduler_strong_APA_Get_self( context );
+  node = _Scheduler_strong_APA_Node_downcast( node_base );
+
+  if( _Chain_Is_node_off_chain( &node->Ready_node ) ) {
+    _Chain_Append_unprotected( &self->Ready, &node->Ready_node );
+  }
+  else {
+    _Chain_Extract_unprotected( &node->Ready_node );
+    _Chain_Set_off_chain( &node->Ready_node );
+    _Chain_Append_unprotected( &self->Ready, &node->Ready_node );
+  }
 }
 
-void _Scheduler_strong_APA_Node_initialize(
-  const Scheduler_Control *scheduler,
-  Scheduler_Node          *node,
-  Thread_Control          *the_thread,
-  Priority_Control         priority
+static inline void _Scheduler_strong_APA_Move_from_scheduled_to_ready(
+  Scheduler_Context *context,
+  Scheduler_Node    *scheduled_to_ready
 )
 {
-  Scheduler_Context            *context;
-  Scheduler_strong_APA_Context *self;
-  Scheduler_strong_APA_Node    *the_node;
+  Priority_Control insert_priority;
 
-  the_node = _Scheduler_strong_APA_Node_downcast( node );
-  _Scheduler_SMP_Node_initialize(
-    scheduler,
-    &the_node->Base,
-    the_thread,
-    priority
-  );
+  if( !_Chain_Is_node_off_chain(  &scheduled_to_ready->Node.Chain ) ) 
```

Comment:
add explicit `{` even for one-line blocks.

Reply:
Added.